### ISSUE TITLE
Allow Image Removal through Parent Component.

### ISSUE THREAD

**Issue by talivodworks** [marmelab/react-admin#5282]
**Is your feature request related to a problem? Please describe.**
I have a use case, let say I want to upload a file and for some reasons my `onDrop` function has gone into error state e.g Server responded with error. Now the Image thumbnail is still in the preview list. There is no way I can remove the file from the list or Identify that an error has occurred for that.

```js
import React, {useState} from 'react';
import { ImageInput, ImageField } from 'react-admin';



export const UploadImage = (props) => {
	const [imageId, setImageId] = useState();
	return (
		<ImageInput
			accept="image/*"
			source="profileImage"
			reference="user"
			multiple={true}
			options={
				{
					onDrop: async ([file]) => {
						// Assume this will fail
						try {
							const id = await uploadToServer(file)
							setImageId(id);
						} catch (e) {
							// Remove Image Preview Here...
						}
					}
				}
			}
		>
			<ImageField source="profileImage" />
		</ImageInput>
	)
}
```

**Describe the solution you'd like**
I have seen this issue [ImageInput keeps adding imagest to state](https://github.com/marmelab/react-admin/issues/349), this works fine. It would be great if we have a way to trigger the removal process through JS (without clicking the remove button)

**Describe alternatives you've considered**
I can't think of any alternative solution to this issue as the state which is keeping the track of file items is maintain inside the [FileInput.tsx](https://github.com/marmelab/react-admin/blob/master/packages/ra-ui-materialui/src/input/FileInput.tsx)
- [beforeRender](https://github.com/marmelab/react-admin/blob/master/packages/ra-ui-materialui/src/input/FileInput.tsx#L118-L130) 
- [InRender](https://github.com/marmelab/react-admin/blob/master/packages/ra-ui-materialui/src/input/FileInput.tsx#L218-L230)

and there is no way to change that state.

**Comment by djhi** [comment]
Please follow the issue template

**Comment by talivodworks** [comment]
> Please follow the issue template

Before I was not sure which template to follow that is why i tried my best to explain this in well formatted way. But I believe it is more of a feature request so I have followed the new feature template.

Thank you.

**Comment by Luwangel** [comment]
You're right, it's between a feature and an issue. So I'm labelling this issue as an enhancement.

Have a good day,
Adrien

**Comment by WiXSL** [comment]
`master` branch is feature-freeze, but this has been implemented already in version 4 #7003.
